lib/src/summation/sum_interface.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// sum_interface.dart
+// Interface for summation and counting.
+//
+// 2024 August 26
+// Author: Max Korbel <max.korbel@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// A [PairInterface] representing an amount and behavior for inclusion in a
+/// sum or count.
+class SumInterface extends PairInterface {
+  /// Whether an [enable] signal is present on this interface.
+  final bool hasEnable;
+
+  /// The [amount] to increment/decrement by, depending on [increments].
+  late final Logic amount =
+      fixedAmount != null ? Const(fixedAmount, width: width) : port('amount');
+
+  /// Controls whether it should increment or decrement (based on [increments])
+  /// (active high).
+  ///
+  /// Present if [hasEnable] is `true`.
+  Logic? get enable => tryPort('enable');
+
+  /// The [width] of the [amount].
+  final int width;
+
+  /// If `true`, will increment. If `false`,  will decrement.
+  final bool increments;
+
+  /// If non-`null`, the constant value of [amount].
+  final dynamic fixedAmount;
+
+  /// Creates a new [SumInterface] with a fixed or variable [amount], optionally
+  /// with an [enable], in either positive or negative direction based on
+  /// [increments].
+  ///
+  /// If [width] is `null`, it can be inferred from [fixedAmount] if provided
+  /// with a type that contains width information (e.g. a [LogicValue]). There
+  /// must be enough information provided to determine the [width].
+  ///
+  /// If a [fixedAmount] is provided, then [amount] will be tied to a [Const]. A
+  /// provided [fixedAmount] must be parseable by [LogicValue.of]. Note that the
+  /// [fixedAmount] will always be interpreted as a positive value truncated to
+  /// [width]. If no [fixedAmount] is provided, then [amount] will be a normal
+  /// [port] with [width] bits.
+  ///
+  /// If [hasEnable] is `true`, then an [enable] port will be added to the
```

Comment:
I do this differently:  if someone passes in an enable in the optional list, then I add it as an input.  This avoids a boolean and gives me the Logic they want to connect as an enable.  So would you prefer to pass in the enable itself (if not passed, don't create an input)

Reply:
The difference in this case is that we're generating an `Interface` object, not a `Module`, and the pattern for these is usually to accept configuration information, not actual `Logic`s, since it's not going to actually use that `Logic` for anything.  Do you still think something is worth changing here?

Reply:
Filed #91 to help track this more generally